### ISSUE TITLE
Modify `Linter` module for aspects

### ISSUE THREAD

**Issue by pareksha** [coala/coala#5694]
`Linter` module needs to be modified so that the `Result` object makes use of aspects. It can be made similar to the implementation in the `Result` module plus more features can be added to make it even more usable.

**Comment by monsij** [comment]
Is anything left to be done for this issue ? @pareksha

**Comment by pareksha** [comment]
Yes @monsij, many coala linters make use of `@linter` decorator which isn't compatible with aspects yet. We cannot migrate those linters into aspects until `linter` module is made compatible.

This is what this issue is about. You can take this up if you're interested. 

You can find some more information in the comments of https://github.com/coala/coala-bears/issues/2657

**Comment by KVGarg** [comment]
@pareksha Do we need to modify Linter class by adding a new argument named `aspect` in ["allowed_options"](https://github.com/coala/coala/blob/master/coalib/bearlib/abstractions/Linter.py#L34)? Or add it in ["supported_names"](https://github.com/coala/coala/blob/master/coalib/bearlib/abstractions/Linter.py#L73) ?

**Comment by pareksha** [comment]
Adding in supported names or allowed options is fine but linter class must make use of aspect arguments provided to it. It would be better to discuss the implementation details in the aspects thread on gitter

**Comment by KVGarg** [comment]
@pareksha I'm struggling over one query, which have put my lot of my time on this issue which I wasn't expecting! Can you help me over it?
Do we need to find `aspect` corresponding to each result using some algorithm or the aspects will itself be provided from `@linter decorator` which we will parse and get aspect corresponding to each result?
Can you help me in going in right direction, if I'm going wrong?
